outlook/src/taskpane/components/Project/TasksSection/SelectProjectDropdown.tsx

```diff
@@ -0,0 +1,171 @@
+import {Spinner, SpinnerSize, TextField} from "office-ui-fabric-react";
+import * as React from "react";
+import Partner from "../../../../classes/Partner";
+import Project from "../../../../classes/Project";
+import {ContentType, HttpVerb, sendHttpRequest} from "../../../../utils/httpRequest";
+import {OdooTheme} from "../../../../utils/Themes";
+import {_t} from "../../../../utils/Translator";
+import api from "../../../api";
+import "./SelectProjectDropdown.css";
+import AppContext from '../../AppContext';
+
+type ProjectDialogProps = {
+    partner: Partner;
+    onProjectClick: (project: Project) => void;
+}
+
+type ProjectDialogState = {
+    query: string;
+    isLoading: boolean;
+    projects: Project[];
+}
+
+class SelectProjectDropdown extends React.Component<ProjectDialogProps, ProjectDialogState> {
+
+    constructor(props, context) {
+        super(props, context);
+        this.state = {query: "", isLoading: false, projects: []};
+    }
+
+    private projectsRequest;
+
+    private onQueryChanged = (event) => {
+        let query = event.target.value;
+        this.setState({query: query});
+        this.cancelProjectsRequest();
+        if (query.length > 0) {
+            this.getProjectsRequest(query);
+        } else {
+            this.setState({isLoading: false, projects: []});
+        }
+    }
+
+    private cancelProjectsRequest = () => {
+        if (this.projectsRequest)
+            this.projectsRequest.cancel();
+    };
+
+    private getProjectsRequest = (searchTerm: string) => {
+
+        if (searchTerm.length > 0) {
+            this.setState({isLoading: true});
+
+            this.projectsRequest = sendHttpRequest(HttpVerb.POST,
+                api.baseURL + api.searchProject, ContentType.Json,
+                this.context.getConnectionToken(), {search_term: searchTerm}, true);
+
+            this.context.addRequestCanceller(this.projectsRequest.cancel);
+
+            this.projectsRequest.promise.then(response => {
+                const parsed = JSON.parse(response);
+                let projects = []
+                if (parsed.result.length > 0)
+                    projects = parsed.result.map(project_json => Project.fromJson(project_json));
+                this.setState({projects: projects, isLoading: false});
+            }).catch((error) => {
+                if (error) {
+                    this.setState({isLoading: false, projects: []});
+                    this.context.showHttpErrorMessage(error);
+                }
+            });
+        }
+
+    }
+
+    private createProject = () => {
+
+        const createProjectRequest = sendHttpRequest(HttpVerb.POST,
+            api.baseURL + api.createProject, ContentType.Json,
+            this.context.getConnectionToken(), {name: this.state.query}, true);
```

Comment:
Can't we forward a cid here for creation, as we now have it available at hand ? Otherwise in case of multi company I fear we get into some strange results with default company not matching cid in outlook ?

Reply:
cids in outlook will contain all existing companies, so I don't think that that would create an issue.